File: ServerProject/src/main/java/com/example/meditime/model/AdherenceLog.java
// Amy Wickham 12178502
package com.example.meditime.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalTime;

/**
 * Entity representing a log of medication adherence.
 * Tracks when a user (e.g., carer) checked the medication adherence
 * for a specific client medication and records the adherence rate.
 */
@Entity
@Getter
@Setter
public class AdherenceLog {

    // Primary key for the adherence log
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long adherenceId;

    // The client medication associated with this adherence log
    @ManyToOne
    private ClientMedication clientMedication;

    // The user (e.g., carer) who logged the adherence
    @ManyToOne
    private User user;

    // The specific medication log entry linked to this adherence (optional)
    @ManyToOne
    @JoinColumn(name = "medication_log_id")
    private MedicationLog medicationLog;

    // The time at which adherence was checked
    private LocalTime checkedTime;

    // Adherence rate recorded at the check time (e.g., percentage or ratio)
    private Double adherenceRate;
}

File: Mobile/app/src/main/java/com/example/medicationapp/model/AdherenceLog.kt


package com.example.medicationapp.model

import java.time.LocalDateTime

/**
 * Entity representing a log of medication adherence by a user.
 *
 * - Logs are linked to a specific medication (ClientMedication) and a user (User).
 * - Stores the time when adherence was checked and the adherence rate as a percentage (0.0 to 1.0).
 * - On deletion of the related medication or user, associated logs are automatically deleted (CASCADE).
 * - Indexed on clientMedicationId and userId to optimize queries filtering by these columns.
 */


data class AdherenceLog(
    val adherenceId: Long,

    val clientMedicationId: Long,
    val userId: Long,
    val checkedTime: LocalDateTime,
    val adherenceRate: Double
)
